daemon/container_unix.go

```diff
@@ -1025,11 +1033,11 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 		container.updateSandboxNetworkSettings(sb)
 	}
 
-	if err := ep.Join(sb); err != nil {
+	if err = ep.Join(sb); err != nil {
 		return err
 	}
```

Comment:
should we also include a defered ep.Leave if the following call to `UpdateJoinInfo` fails